src/meta/api/src/kv_api_utils.rs

```diff
@@ -414,3 +418,91 @@ pub fn get_share_database_id_and_privilege(
         ShareHasNoGrantedDatabase::new(&name_key.tenant, &name_key.share_name),
     )))
 }
+
+// Return true if all the database data has been removed.
+pub async fn is_all_db_data_has_been_removed(
+    kv_api: &(impl KVApi + ?Sized),
+    db_id: u64,
+) -> Result<bool, MetaError> {
+    let dbid = DatabaseId { db_id };
+
+    let (db_meta_seq, db_meta): (_, Option<DatabaseMeta>) = get_struct_value(kv_api, &dbid).await?;
+    if db_meta_seq != 0 || db_meta.is_some() {
+        return Ok(false);
+    }
```

Comment:
Checking `db_meta_seq` would be enough. `db_meta_seq == 0` and `db_meta` is `Some` is a bug.

```suggestion
    debug_assert_eq!((db_meta_seq == 0), db_meta.is_none());
    if db_meta_seq != 0 {
        return Ok(false);
    }
```